### PR TITLE
Add assigned workshop and registered workshop to facilitator cohort view

### PR DESCRIPTION
**Pull request by aoby**
…, clean up associated code and tests, and add tests

### CODE REVIEW

File: dashboard/app/models/pd/application/facilitator1819_application.rb [hunk 1]
```diff
@@ -505,10 +509,6 @@ def destroy_fit_autoenrollment
       self.auto_assigned_fit_enrollment_id = nil
     end
 
-    def fit_workshop
-      Pd::Workshop.find(fit_workshop_id) if fit_workshop_id
-    end
-
```
Comment (aoby) [inline dashboard/app/models/pd/application/facilitator1819_application.rb:511]: This was a [duplicate](https://github.com/code-dot-org/code-dot-org/blob/810bd1beb9505ccb539f77b29817a4f908ee3964/dashboard/app/models/pd/application/facilitator1819_application.rb#L82-L84)

File: dashboard/app/models/pd/application/teacher1819_application.rb [hunk 2]
```diff
@@ -536,16 +536,8 @@ def principal_approval
...
-    def assigned_workshop
-      pd_workshop_id ? Pd::Workshop.find(pd_workshop_id).date_and_location_name : ''
-    end
-
-    def registered_workshop
-      if pd_workshop_id
-        Pd::Enrollment.exists?(pd_workshop_id: pd_workshop_id, user: user) ? 'Yes' : 'No'
-      else
-        ''
-      end
```
Comment (aoby) [inline dashboard/app/models/pd/application/teacher1819_application.rb:548]: I moved this display logic into the serializers, and added `registered_workshop?` into the `WorkshopAutoenrolledApplication` base class (shared between teacher and facilitator applications).

File: dashboard/app/models/pd/application/teacher1819_application.rb [hunk 3]
```diff
@@ -536,16 +536,8 @@ def principal_approval
...
+    def date_accepted
```
Comment (mehalshah) [inline dashboard/app/models/pd/application/teacher1819_application.rb]: Can this be pushed up to the application_base? accepted_at is a db column

File: dashboard/test/models/pd/application/facilitator1819_application_test.rb [hunk 4]
```diff
@@ -6,6 +6,13 @@ class Facilitator1819ApplicationTest < ActiveSupport::TestCase
     self.use_transactional_test_case = true
     setup_all do
       @regional_partner = create :regional_partner
+      @workshop = create :pd_workshop
+      @application = create :pd_facilitator1819_application
+      @application_with_fit_workshop = create :pd_facilitator1819_application, fit_workshop_id: @workshop.id
+    end
+    setup do
+      @application.reload
+      @application_with_fit_workshop.reload
```
Comment (aoby) [inline dashboard/test/models/pd/application/facilitator1819_application_test.rb:15]: Most of this is just test optimization. We were creating applications in many test cases. Here create it once and reuse.

**Comment by aoby** [comment]
[Add “Assigned workshop” and “Registered Workshop” to facilitator Cohort View](https://trello.com/c/VRnRQiqK/471-add-assigned-workshop-and-registered-workshop-to-facilitator-cohort-view)

File: dashboard/app/serializers/api/v1/pd/cohort_view_serializer_base.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    :district_name,
+    :school_name,
+    :email,
+    :assigned_workshop,
+    :registered_workshop,
+    :status
+  )
+
+  def date_accepted
+    object.accepted_at.try(:strftime, '%b %e')
```
Comment (mehalshah) [inline dashboard/app/serializers/api/v1/pd/cohort_view_serializer_base.rb]: If you put accepted_at at the application_base layer, you can get rid of this funciton

File: dashboard/app/serializers/api/v1/pd/facilitator_application_cohort_view_serializer.rb [hunk 6]
```diff
@@ -1,7 +1,17 @@
-class Api::V1::Pd::FacilitatorApplicationCohortViewSerializer < ActiveModel::Serializer
-  attributes :id, :date_accepted, :applicant_name, :district_name, :school_name, :email, :status
+module Api::V1::Pd
```
Comment (mehalshah) [inline dashboard/app/serializers/api/v1/pd/facilitator_application_cohort_view_serializer.rb:1]: I like this

File: dashboard/app/models/pd/application/teacher1819_application.rb (hunk 3 above)
Comment (aoby) [reply dashboard/app/models/pd/application/teacher1819_application.rb]: D'oh, it's already [there](https://github.com/code-dot-org/code-dot-org/blob/547eb6860fe20b1f75afd4b8bd9e814e2c6890eb/dashboard/app/models/pd/application/application_base.rb#L297-L299).

Comment (aoby) [reply dashboard/app/models/pd/application/teacher1819_application.rb]: After offline discussion, we agreed this abbreviated date format (e.g. "Mar 9") is too display-specific to belong in the model. It's also not sortable, either in the UI table or csv export. I'm going to change the format to [ISO 8601](https://xkcd.com/1179/) and re-format to abbreviated month and day in the UI on the client.

**Pull request merged**